### PR TITLE
UHF-4528: Add template for minimal display on TPR-units

### PR DESCRIPTION
**Pull request by teroelonen**
# Add a way to list large amounts of units on a listing [UHF-4528](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4528) & [UHF-4532](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4532) 
Make it possible for the content producers to tag and list large numbers of units at the same time (for example daycare units).

## What was done
* Added template for the minimal display on units.

## How to install
* Check from the PR https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/279

## How to test
* Check from the PR https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/279

## Other PRs
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/279
* https://github.com/City-of-Helsinki/drupal-helfi-kasvatus-koulutus/pull/116

### CODE REVIEW

**Review by tapsal — APPROVED** [review]
Great! 🚀

**Review by Arkkimaagi — CHANGES_REQUESTED** [review]

File: templates/module/helfi_tpr/tpr-unit.html.twig [hunk 1]
```diff
@@ -280,4 +280,28 @@
 
   {% include '@hdbt/module/helfi_tpr/tpr-content-liftup.html.twig' %}
 
+{% elseif view_mode == 'minimal' %}
+
+  <div class="unit unit--minimal">
+    <a href="{{ url('entity.tpr_unit.canonical', { 'tpr_unit': entity.id() }) }}" class="unit__link">
+      <div class="unit__text">
+        <div class="unit__text__content">
+          <h3 class="unit__title">
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply templates/module/helfi_tpr/tpr-unit.html.twig]: We don't - changed it to default to h3 and if you decide to use this display outside a component container (that has title that defaults to h2) which we haven't done so far you need to extend this template and give it some other level heading element. Refactored the teaser to have the same.

File: src/scss/06_components/paragraphs/_unit-search.scss [hunk 2]
```diff
@@ -150,3 +171,22 @@
...
+  margin-top: $spacing-and-half;
+}
+
+.unit-search__no-results-desc {
+  @include font('body');
+}
+
+// Unit search with large volume of units listed such as daycare search.
+.views--unit-search--large-volume {
+  .views-row {
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply src/scss/06_components/paragraphs/_unit-search.scss]: Yeah can be on single row. Just a mistake - had some other stuff under it before and forgot to remove nesting after it became redundant.

File: src/scss/06_components/paragraphs/_unit-search.scss [hunk 3]
```diff
@@ -116,6 +116,27 @@
...
+
+  &.unit-search__address-not-found {
+    font-weight: bold;
+    padding-left: $spacing-double;
+    position: relative;
+
+    &::before {
+      @include pseudo-icon('alert-circle', $spacing-and-half);
+      left: 0;
+      margin-top: -$spacing-and-half * 0.5;
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply src/scss/06_components/paragraphs/_unit-search.scss]: Hahahaha I sure agree - this was copied from your code so I thought maybe you don't want me to write 12 here but calculate it from the the line height of the body. Changed to -12px and added a comment.

File: src/scss/06_components/pages/_unit.scss [hunk 4]
```diff
@@ -548,3 +548,106 @@
...
+      margin-top: 0;
+    }
+
+    .hel-icon {
+      --icon-size: #{$spacing-double};
+      @include hover-link-arrow-transition();
+    }
+  }
+
+  &:hover {
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply src/scss/06_components/pages/_unit.scss]: Added and changed the style to apply to correct place - the link and not the surrounding element.

File: src/scss/06_components/pages/_unit.scss [hunk 5]
```diff
@@ -548,3 +548,106 @@
...
+      width: 100%;
+    }
+  }
+
+  .unit__text__content {
+    @include breakpoint($breakpoint-m) {
+      display: flex;
+      justify-content: space-between;
+      // Width of the element - (icon width + icon padding on left)
+      width: calc(100% - (#{$spacing-double} + #{$spacing-and-half}));
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply src/scss/06_components/pages/_unit.scss]: Parenthesis never hurt anyone!! :D It was my motto in math at school so maybe thats why I used them there. Anyway removed them if they made it less readable.

**Review by Arkkimaagi — APPROVED** [review]
Code looks good to go!

**Pull request merged**